### ISSUE TITLE
Support custom configuration file

### ISSUE THREAD

**Issue by Ailrun** [siddharthkp/bundlesize#225]
**Do you want to request a *feature* or report a *bug*?**
Request a feature

**What is the current behavior?**
Read the `package.json`

**If this is a feature request, what is motivation or use case for changing the behavior?**
I want to separate configs into specific directory.

**Please mention other relevant information.**
- node version: 9.11.1
- npm version: 6.0.0
- Operating system: Manjaro
- bundlesize version: 0.17.0
- CI you are using: TravisCI

**Comment by trevordmiller** [comment]
Agreed. It would be nice to have a `.bundlesizerc` or `.bundlesizerc.json` file option for those who like to keep config out of `package.json` (following other tools: `.babelrc`, `.eslintrc.json`, `.lintstagedrc.json`, `.prettierrc.json`, `now.json`, etc.).

Thanks for the great tool!

**Comment by trevordmiller** [comment]
For example, right now I have an npm script that looks like this:

```js
"broken-links": "blc http://yoursite.com -ro --exclude=linkedin.com --exclude=visualstudio.com --exclude=netflix.com --exclude=apple.com"
```

It would be nice if I could move all the config to a `.bundlesizerc.json` configuration file like:

```js
{
  "input": "https://yoursite.com",
  "recursive": true,
  "ordered": true,
  "exclude": [
    "linkedin.com",
    "visualstudio.com",
    "netflix.com",
    "apple.com"
  ]
}
```

Then the npm script could simply be:

```js
"broken-links": "blc"
```

**Comment by Aghassi** [comment]
Adding this should be very straight forward. https://github.com/davidtheclark/cosmiconfig can be added as a dependency, and it takes very few lines of code. @siddharthkp, I can take this on unless there is any objection to using this package.

**Comment by trevordmiller** [comment]
That would be awesome. cosmiconfig is what many other tools use. Prettier, for example: https://prettier.io/docs/en/configuration.html

**Comment by ssulivan11** [comment]
Have a custom config setting in this PR:
https://github.com/siddharthkp/bundlesize/pull/212

In our monorepo we maintianed hundreds of individual bundles and the package.json got ugly 😊

**Comment by siddharthkp** [comment]
Implemented in https://github.com/siddharthkp/bundlesize/pull/316 and published in 0.18.0 🎉
